.github/workflows/create-gh-release.yml

```
@@ -0,0 +1,59 @@
+name: Create GH Release
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+  workflow_dispatch:
+
+# Allow only one concurrent deployment,but do NOT cancel in-progress runs as
+# we want to allow these release deployments to complete.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: false
+
+jobs:
+  create-release:
+    # a gh release will be created in the following conditions:
+    # 1. if it is a manual trigger (workflow_dispatch)
+    # 2. if the workflow was triggered by a merged changeset PR
+    if: |
+      (github.event_name == 'workflow_dispatch' ||
+      (github.event.pull_request.merged == true &&
+        (github.event.pull_request.head.ref == 'changeset-release/main' ||
+        contains(github.event.pull_request.labels.*.name, 'automated-release'))))
+    name: Create GH Release
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        with:
+          node-version: 20
+          registry-url: https://registry.npmjs.org/
+
+      - name: Check if GitHub repo package version is latest
+        run: |
+          # get any package version from the repo
+          cd packages/protocol
```

```suggestion
          cd packages/protocol # assumption that all packages have the same version
```

This is a safe assumption for now, but it may change eventually

---

Good call! When it changes, we will need to define how we are going to be tagging the GH releases too I think.